libs/sdk/src/http.rs

```diff
@@ -20,32 +20,42 @@ pub fn prepare_get_request(
     forward_url: &str,
     params: Option<&[QueryParam<'_, '_>]>,
     headers: Option<&[HeaderParam<'_, '_>]>,
-    timeout_secs: u64,
+    timeout_secs: Option<u64>,
 ) -> Result<Request> {
-    let base_url = "http://127.0.0.1:3000";
+    let mut req = Request::builder();
+
     let forward_url_and_params = match params {
         Some(p) => Url::parse_with_params(forward_url, p)?,
         None => Url::parse(forward_url)?,
     };
 
-    let mut req = Request::builder();
-    req.method(Method::Post);
-    req.uri(base_url);
-
     if let Some(hdrs) = headers {
         for (key, value) in hdrs {
             req.header(*key, *value);
         }
     }
 
-    let payload = Params {
-        endpoint_url: forward_url_and_params.to_string(),
-        seconds: timeout_secs,
-        request_method: "GET".to_string(),
-        request_body: None,
-    };
+    match timeout_secs {
+        Some(timeout_secs) => {
+            req.method(Method::Post);
+            req.uri("http://127.0.0.1:3000");
+
+            let payload = Params {
+                endpoint_url: forward_url_and_params.to_string(),
+                seconds: timeout_secs,
+                request_method: "GET".to_string(),
+                request_body: None,
+            };
 
-    req.body(serde_json::to_string(&payload).unwrap());
+            req.body(serde_json::to_string(&payload).unwrap());
+        }
+        _ => {
```

Comment:
nit: Don't be so cheap :smile:
```suggestion
        None => {
```